src/Draco.Compiler/Api/CodeCompletion/CompletionItem.cs

```diff
@@ -0,0 +1,32 @@
+using System.Collections.Immutable;
+using Draco.Compiler.Api.Semantics;
+using Draco.Compiler.Api.Syntax;
+
+namespace Draco.Compiler.Api.CodeCompletion;
+
+/// <summary>
+/// Represents a code completion item.
+/// </summary>
+/// <param name="Change">The <see cref="TextChange"/> this item provides.</param>
+/// <param name="Symbols">All <see cref="ISymbol"/>s representing this completion (usually symbol representing type etc. or multiple <see cref="FunctionSymbol"/>s representing an overload).</param>
+/// <param name="Kind">The <see cref="CompletionKind"/> of this completion.</param>
+/// <param name="Context">The <see cref="CompletionContext"/> of this completion.</param>
+public record class CompletionItem(TextChange Change, ImmutableArray<ISymbol> Symbols, CompletionKind Kind, params CompletionContext[] Context)
```

Comment:
Does the completion item need to carry the context? It feels like to me this is something the provider would be concerned with.